### PR TITLE
MVEL to SOLR translation enhancements

### PR DESCRIPTION
**Pull request by oleksiimiroshnyk**
BroadleafCommerce/QA#4129
MVEL to SOLR translation enhancements

### CODE REVIEW

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java [hunk 1]
```diff
@@ -42,137 +52,290 @@
...
-    protected String getCustomFieldPropertyName(String mvelRule) {
-        int startIndex = mvelRule.indexOf("[\"") + 2;
-        int endIndex = mvelRule.lastIndexOf("\"]");
+    // Expected Category Rule Format - CollectionUtils.intersection(product.?allParentCategoryIds,["2002"]).size()>0
+    protected boolean isCategoryTargetingRule(String mvelRule) {
+        return mvelRule.matches(CATEGORY_FORMAT_REGEX);
+    }
 
-        return mvelRule.substring(startIndex, endIndex);
+    // Expected Category Rule Format - CollectionUtils.intersection(product.?allParentCategoryIds,["2002"]).size()>0
```
Comment (danielcolgrove) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java]: Can we extend this to support multiple categories?  Any of the CollectionUtils should support multiple values
CollectionUtils.intersection(product.?allParentCategoryIds,["2002","2007"]).size()>0

**Review by danielcolgrove — CHANGES_REQUESTED** [review]

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java (hunk 1 above)
Comment (oleksiimiroshnyk) [reply core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java]: should be supported now

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/domain/SearchCriteria.java [hunk 2]
```diff
@@ -20,8 +20,7 @@
 import org.broadleafcommerce.core.catalog.domain.Category;
 import org.broadleafcommerce.core.search.service.solr.SolrHelperService;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/domain/SearchCriteria.java]: Revert this change and update the code formatter in your IDE to not squash imports into a wildcard import in order to be consistent with Broadleaf's code formatting guidelines

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java [hunk 3]
```diff
@@ -18,161 +18,382 @@
...
 
-import java.util.List;
-
 import javax.annotation.Resource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java]: Update your IDE to organize imports as org, com, java, javax to be consistent with Broadleaf's formatting style. We try to be consistent with this as to reduce merge conflicts as much as possible

**Review by cja769 — CHANGES_REQUESTED** [review]
I don't know enough about this to comment on the implementation however other than the import changes the code looks fine to me as long as as it solves the issue at hand with at any regression issues.
